command/testing.go

```diff
@@ -151,11 +149,50 @@ func RunCommand(args string) (*cmdOut, error) {
 		}
 	})
 
+	return rootCmd, cmd, rootArgv, argv, func() {
+		cmd.SetOut(nil)
+		cmd.SetErr(nil)
+	}, nil
+}
+
+func RunCommand(args string) (*cmdOut, error) {
+	cmdOutStreams := cmdOut{
+		outBuf: &bytes.Buffer{},
+		errBuf: &bytes.Buffer{},
+	}
+
+	rootCmd, _, rootArgv, _, cleanUpFunc, err := setupCommand(args, &cmdOutStreams)
+	if err != nil {
+		return nil, err
+	}
+	defer cleanUpFunc()
+
+	rootCmd.SetArgs(rootArgv)
+
 	_, err = rootCmd.ExecuteC()
-	cmd.SetOut(nil)
-	cmd.SetErr(nil)
 
-	return &cmdOut{&outBuf, &errBuf}, err
+	return &cmdOutStreams, err
+}
+
+func PrepareCommandArguments(command string) (*cobra.Command, []string, *cmdOut, func(), error) {
```

Comment:
This is for setting up the `cobra.Command` and the `args []string`, to be directly passed to our functions, so that we can invoke them directly, without relying on cobra to do that.